src/examples/zkapps/voting/test.ts

```diff
@@ -31,43 +42,640 @@ export async function testSet(
+      );
+      voting.vote(fakeCandidate, votersStore.get(0n)!);
+      voting.sign(votingKey);
+    },
+    feePayer,
+    'assert_equal'
+  );
+
+  console.log('unregistered voter attempting to vote');
+
+  await assertValidTx(
+    false,
+    () => {
+      let fakeVoter = Member.from(
+        PrivateKey.random().toPublicKey(),
+        Field.zero,
+        UInt64.from(50)
+      );
+      voting.vote(fakeVoter, votersStore.get(0n)!);
+      voting.sign(votingKey);
+    },
+    feePayer,
+    'assert_equal'
+  );
+
+  console.log('attempting to vote for voter...');
+
+  await assertValidTx(
+    false,
+    () => {
+      const voter = votersStore.get(0n)!;
+
+      voting.vote(voter, votersStore.get(0n)!);
+      voting.sign(votingKey);
+    },
+    feePayer,
+    'assert_equal'
+  );
+
+  /*
+    test case description:
+      happy path - vote counting
+
+    preconditions:
+      - votes were emitted
+
+    tested cases:
+      - counting votes -> success
+
+    expected results:
+      - counts all emitted votes through sequence events
+      - updates on-chain state to equal off-chain state
+      - prints final result (helper function)
+
+  */
+  console.log('counting votes...');
+
+  await assertValidTx(
+    true,
+    () => {
+      voting.countVotes();
+      voting.sign(votingKey);
+    },
+    feePayer
+  );
+
+  if (!voting.committedVotes.get().equals(votesStore.getRoot()).toBoolean()) {
+    throw Error(
+      'votesStore merkle root does not match on-chain committed votes'
     );
-    voting.accumulatedVotes.set(Experimental.Reducer.initialActionsHash);
-  });
-  tx.send();
+  }
+
+  console.log('election is over, printing results');
+  printResult(voting, votesStore);
+
+  console.log('testing after election state');
+
+  /*
+    test case description:
+      registering voter and candidates AFTER election has ended
+    
+    preconditions:
+      - election ended
+
+    tested cases:
+      - registering voter -> failure
+      - registering candidate -> failure
+
+    expected results:
+      - no state changes
+      - no events emitted
+
+  */
+  console.log('attempting to register voter after election has ended');
+
+  await assertValidTx(
+    false,
+    () => {
+      let voter = Member.from(
+        PrivateKey.random().toPublicKey(),
+        Field.zero,
+        params.voterPreconditions.minMina.add(1)
+      );
+      voting.voterRegistration(voter);
+      voting.sign(votingKey);
+    },
+    feePayer,
+    'Expected'
+  );
+
+  console.log('attempting to register candidate after election has ended');
 
-  console.log('all contracts deployed');
+  await assertValidTx(
+    false,
+    () => {
+      let candidate = Member.from(
+        PrivateKey.random().toPublicKey(),
+        Field.zero,
+        params.candidatePreconditions.minMina.add(1)
+      );
+      voting.candidateRegistration(candidate);
+      voting.sign(votingKey);
+    },
+    feePayer,
+    'Expected'
```

Comment:
For this and similar cases above. 
In real world I'd not only check the presence of single word in error message.
But probably `Ok` for this case.